### PR TITLE
Add a basic serial controller that receives and sends back a serial message from Arduino

### PR DESCRIPTION
**Pull request by andrew103**
Right now its in POC state so very un-ideal setup, but I'll update it later.

### CODE REVIEW

File: trajectory_tracker/tracker.py [hunk 1]
```diff
@@ -19,6 +21,17 @@ def __init__(self):
 
     timer_period = 2 # seconds
     self.timer = self.create_timer(timer_period, self.timer_callback)
+    self.ard_check = self.create_timer(0.1, self.ard_check_callback)
+
+    self.arduino_port = serial.Serial(
+      port="/dev/ttyAMA1",
```
Comment (agupta231) [inline trajectory_tracker/tracker.py:27]: can this be dynamically computed? and/or make this into a cli flag. Totally cool if you don't want to do it it, but def leave a `TODO` comment

File: trajectory_tracker/tracker.py [hunk 2]
```diff
@@ -19,6 +21,17 @@ def __init__(self):
...
+      baudrate=115200
+    )
+
+  def ard_check_callback(self):
```
Comment (agupta231) [inline trajectory_tracker/tracker.py:31]: not gonna be anal about it now, but we should be strict about docstrings and type hints

File: trajectory_tracker/tracker.py [hunk 3]
```diff
@@ -19,6 +21,17 @@ def __init__(self):
...
+    if self.arduino_port.in_waiting > 0:
```
Comment (agupta231) [inline trajectory_tracker/tracker.py:32]: don't need the `>0` cause of implicit python booleans. Can be changed to `if self.arudino_port.in_waiting:`

File: trajectory_tracker/tracker.py [hunk 4]
```diff
@@ -19,6 +21,17 @@ def __init__(self):
...
+      received_data = self.arduino_port.read(1)
+      self.arduino_port.write(received_data)
```
Comment (agupta231) [inline trajectory_tracker/tracker.py:34]: so is this thing just reading and spitting back the same data?

**Review by agupta231 — CHANGES_REQUESTED** [review]
Couple of small code health changes. Most work should just be tossed into a todo.

File: trajectory_tracker/tracker.py (hunk 4 above)
Comment (andrew103) [reply trajectory_tracker/tracker.py:34]: For now, yea that's all it does.

File: trajectory_tracker/tracker.py (hunk 1 above)
Comment (andrew103) [reply trajectory_tracker/tracker.py:27]: For Raspis, the first usable serial port that isn't the serial console will be ttyAMA1. Only reason to have it variable would be if you needed to specify a different port for use with the gym environment. If that's the case, I'll make it a cli flag

File: trajectory_tracker/tracker.py (hunk 4 above)
Comment (agupta231) [reply trajectory_tracker/tracker.py:34]: sg

File: trajectory_tracker/tracker.py (hunk 1 above)
Comment (agupta231) [reply trajectory_tracker/tracker.py:27]: hmmm I'll need to look into that, and like it's fine now cause it'll take 2 seconds to change in the future.

Comment (andrew103) [reply trajectory_tracker/tracker.py:27]: Ok. I'll put a `TODO` as a reminder to come back to it

**Review by agupta231 — APPROVED** [review]
lgtm

**Pull request merged**